src/Htmx/HtmxResponseHeaders.cs

```diff
@@ -153,5 +161,78 @@ public HtmxResponseHeaders Retarget(string value)
             _headers[Keys.Retarget] = value;
             return this;
         }
+
+        /// <summary>
+        /// Trigger a client side event named <paramref name="eventName"/> at different stages of the HTMX lifecycle.
+        /// </summary>
+        /// <param name="eventName">Event name</param>
+        /// <param name="detail">Detail to be sent with the event, string or other JSON mappable object</param>
+        /// <param name="timing">The HTMX request lifecycle step to run the trigger at</param>
+        /// <returns></returns>
+        public HtmxResponseHeaders WithTrigger(string eventName, object? detail = null,
+            HtmxTriggerTiming timing = HtmxTriggerTiming.Default)
+        {
+            if (!_triggers.ContainsKey(timing))
+            {
+                _triggers.Add(timing, new() { {eventName, detail ?? string.Empty } });
+            }
+            else
+            {
+                _triggers[timing].Add(eventName, detail ?? string.Empty);
+            }
+
+            return this;
+        }
+        
+        /// <summary>
+        /// Process the Triggers
+        /// </summary>
+        internal HtmxResponseHeaders Process()
+        {
+            if (_triggers.ContainsKey(HtmxTriggerTiming.Default))
+            {
+                if (_headers.ContainsKey(Keys.Trigger))
+                {
+                    throw new Exception("You must use either WithTrigger(..) or Trigger(..), but not both.");
+                }
+                
+                _headers[Keys.Trigger] = BuildTriggerHeader(HtmxTriggerTiming.Default);
+            }
+            if (_triggers.ContainsKey(HtmxTriggerTiming.AfterSettle))
+            {
+                if (_headers.ContainsKey(Keys.TriggerAfterSettle))
```

Comment:
Maybe add some tests for the unhappy paths as well?

Reply:
Consider it done